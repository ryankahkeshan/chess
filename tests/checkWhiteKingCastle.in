setup
+ K e1 + k e8
+ R h1
+ b d4
+ N e2
= black
done game human human
move d4 c3

move e1 g1
move e2 c3
move e8 f7
move e1 g1
resign
