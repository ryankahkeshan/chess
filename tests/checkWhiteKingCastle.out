8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _R

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ b _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _R

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ b _ _
3 _ _ _ _ 
2  _ _N_ _
1 _ _ K _R

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ b _ _
3 _ _ _ _ 
2  _ _N_ _
1 _ _ K _R

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ b _ _ 
2  _ _N_ _
1 _ _ K _R

  abcdefgh

White is in check.
Invalid move
8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ N _ _ 
2  _ _ _ _
1 _ _ K _R

  abcdefgh

8  _ _ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ N _ _ 
2  _ _ _ _
1 _ _ K _R

  abcdefgh

8  _ _ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ N _ _ 
2  _ _ _ _
1 _ _ _RK 

  abcdefgh

Black is in check.
White wins!
Final Score:
White: 1
Black: 0
